src/GTP.h

```diff
@@ -29,6 +29,49 @@
 #include "GameState.h"
 #include "UCTSearch.h"
 
+struct MoveToAvoid {
+    int color;
+    size_t until_move;
+    int vertex;
+
+    MoveToAvoid(int color, size_t until_move, int vertex)
+        : color(color), until_move(until_move), vertex(vertex)
+    {}
+
+    bool operator==(const MoveToAvoid other) const {
+        return color == other.color &&
+            until_move == other.until_move && vertex == other.vertex;
+    }
+};
+
+class AnalyzeTags {
+public:
+    bool m_invalid;
+    std::vector<MoveToAvoid> m_moves_to_avoid, m_moves_to_allow;
+    int m_interval_centis;
+    int m_who;
+
+    AnalyzeTags() {
+        clear();
```

Comment:
You could use uniform initialization here, i.e. make this = default, and just do

```
public:
    bool m_invalid{true};
    etc..
```

This is currently used by most of the other code. Do away with clear, and just assign from an empty instance when clearing is needed.

This is a matter of taste but I think it would end up slightly cleaner.